### PR TITLE
Replace non-printable chars for the right ones

### PR DESCRIPTION
**Pull request by carla-at-wiris**
## Description

We use multi-byte characters to name some of our variables. Those characters were compromised on previous releases and turned into the following char: '�'.

This PR changes all the occurrences of the non-printable characters inside the `integration` folder.

## Steps to reproduce

1. Follow the instructions on **how to connect to a moodle database** in the `wiris-moodle-docker` project.
2. Start a new moodle with `wiris-moodle-docker`.
3. Replace the content from the `wiris filter` folder inside the new moodle instance, for the content of this project.
4. Create a new quizz and insert a formula on it.
5. Open the database and retrieve the information from the `m_quiz` row by executing:

    ```sql
    SELECT * FROM m_quiz;
    ```
6. Verify that the formula information is stored properly and that there are no '�' characters.
 
---

[#taskid 18388](https://wiris.kanbanize.com/ctrl_board/2/cards/18388/details/)

## Description 2

Wiris Filter is not compatible with Moodle > 3.10 and PHP 8.0. This task also contains the changes to make it compatible.

## Steps to reproduce 

1. Start a new moodle with `wiris-moodle-docker`. Moodle version 3.11 or 4.0, and PHP version 8.0.
2. Replace the content from the `wiris filter` folder inside the new moodle instance, for the content of this project.
3. Go to site administration and verify that it does not appear any error.

---

[#taskid 28300](https://wiris.kanbanize.com/ctrl_board/2/cards/28300/details/)

### CODE REVIEW

**Review by icaparros-at-wiris — APPROVED** [review]

File: integration/WIRISplugins.js [hunk 1]
```diff
@@ -2261,21 +2261,12 @@ com.wiris.system.JsDOMUtils.addStylesheet = function(d,url) {
 	}
 }
 com.wiris.system.JsDOMUtils.loadTextFile = function(elem,func) {
-	com.wiris.system.JsDOMUtils.loadFileImpl(elem,func);
-}
-com.wiris.system.JsDOMUtils.loadPNGImageFile = function(elem,func) {
-	com.wiris.system.JsDOMUtils.loadFileImpl(elem,func,true,"image/png");
-}
-com.wiris.system.JsDOMUtils.loadFileImpl = function(elem,func,asDataURL,accept) {
-	if(asDataURL == null) asDataURL = false;
```
Comment (xripoll-at-wiris) [inline integration/WIRISplugins.js:2270]: I see a lot of content changes in the JS files that go beyond fixing the encoding. Do you know where this come from?

File: integration/lib/Type.class.php [hunk 2]
```diff
@@ -79,7 +79,7 @@ static function createInstance($cl, $args) {
 		if($c === null) {
 			return null;
 		}
-		return $inst = $c->getConstructor() ? $c->newInstanceArgs($args->�a) : $c->newInstanceArgs();
+		return $inst = $c->getConstructor() ? $c->newInstanceArgs($args->»a) : $c->newInstanceArgs();
```
Comment (xripoll-at-wiris) [inline integration/lib/Type.class.php:82]: Why is this one `Â»a` and not `»a`?

**Review by xripoll-at-wiris — COMMENTED** [review]
The task description only talks about encoding issues in our source code/transpiled code. But on the PR instructions you describe how to check that our *data* is properly encoded. How does one affect the other? Had we also previously identified that our data was mangled, besides the code itself?

File: integration/lib/Type.class.php (hunk 2 above)
Comment (xripoll-at-wiris) [reply integration/lib/Type.class.php:82]: @carla-at-wiris do you know if this is relevant?

Comment (carla-at-wiris) [reply integration/lib/Type.class.php:82]: I don't know, but looking at previous commits and other repositories using the same file, this was the `char` used.

**Review by icaparros-at-wiris — APPROVED** [review]

**Pull request merged**